File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
import sys
input = sys.stdin.readline

# 설탕 N
# 봉지 3, 5
# 최대한 적은 봉지

N = int(input())
cnt = 0


# 5로 최대로 나누었을 때, 나머지가 3의 배수이면 출력
a, b = divmod(N, 5)
if b % 3 == 0:
    print(a + b//3)
# 나머지가 3의 배수가 아니라면, 5로 나눈 몫을 하나씩 빼면서 검사
else:
    while a > 0:
        a -= 1
        b += 5
        if b % 3 == 0:
            print(a + b // 3)
            break
    else:
        print(-1)

File: 백준/Silver/2839. 설탕 배달/설탕 배달.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

let n = Number(input[0]);
let five = Math.floor(n / 5);
let answer = 5001;

for (let i = five; i >= 0; i--) {
  let x = n - 5 * i;
  if (x % 3 === 0) {
    answer = Math.min(answer, i + Math.floor(x / 3));
  }
}

if (answer === 5001) console.log(-1);
else console.log(answer);
